tests/meson.build

```diff
@@ -0,0 +1,74 @@
+# Test sources
+test_src = files(
+    'test_CommChannel.cc',
+    'test_Context.cc',
+    'test_exceptions.cc',
+    'test_Executor.cc',
+    #'tests/test_format.cc' needs fmt{} library
+    'test_LockedQueue.cc',
+    'test_main.cc',
+    'test_Message.cc',
+    'test_Sequence.cc',
+    'test_SequenceManager.cc',
+    'test_serialize_sequence.cc',
+    'test_Step.cc',
+    'test_time_types.cc',
+    'test_VariableName.cc',
+)
+
+# The tests are executed in the build dir to avoid pollution of the git repository with
+# the test executable's output files
+test('all',
+    executable(meson.project_name() + '_test',
+        test_src,
+        dependencies : taskolib_dep,
+    ),
+    workdir : meson.current_build_dir(),
+    timeout : 10,
+)
+
+fmt_dep = dependency('fmt', required : false)
+if fmt_dep.found()
+    test('format',
+        executable(meson.project_name() + '_format_test',
+            [ 'test_format.cc', 'test_main.cc', ],
+            dependencies : [ taskolib_dep, dependency('fmt'), ],
+        ),
+        timeout : 10,
+    )
+endif
+
+######
+# Test if all headers are self-contained (Core Guidelines SF.11)
+# This is automated over all public_headers
+
+message('Generating self-containment tests')
+all_header_checks = [ ]
+loop_number = 0
+foreach header_to_check : public_headers
```

Comment:
From description #31 it seem to be complex stuff to validate self-contained header files, isn't it? Wouldn't it be enough to include in all test sources `#include "taskolib/taskolib.h"`? If the corresponding header cannot be resolved we get a direct compilation error. But it only checks the header `#include "taskolib/taskolib.h"` 🤔...

And how does it look like when a header is missing? I mean the concert error message?

Reply:
Well, if you include `taskolib/taskolib.h`, what does that say about `taskolib/Step.h`?
This does not test/solve anything?

Self contained means that you can include that header on its own, as single include, and it brings all the other stuff it needs. So for example if you include `Sequence.h`, that will automatically also include `Step.h`, and you do not need to spell that out in your own `cc` file.

It might look complicated, but on the other hand it is guaranteed that we check all headers we will ever ship to the user and no header will be forgotten if we add new ones.

For example ... lets assume `Step.h` always needs `stdio.h`. Usually you get `stdio.h` from 'everywhere', so potentially we will never notice that the inclusion of `stdio.h` is missing from `Step.h`. All places we **use** `Step.h` also include some standard stuff and that in turn includes `stdio.h`. (Well, this is not real, just as example.)

But then someone comes around it that person wants to just use `Step`s, and includes `Step.h` and ... it does not compile because `Step.h` needs `stdio.h` without including it on its own.

> And how does it look like when a header is missing? I mean the concert error message?

Just a normal compiler error while building the dependencies of `_ignore_me.a`.

Reply:
> it seem to be complex stuff to validate self-contained header files

Well, in part it is 'complicated' because it sets up an abstract mechanism to automatically check **all** public header files, without prior knowledge which files that might be.